photon-ml/src/main/scala/com/linkedin/photon/ml/util/Utils.scala

```diff
@@ -70,11 +70,10 @@ protected[ml] object Utils {
     * @return The feature name
     */
   def getFeatureNameFromKey(key: String, delimiter: String = GLMSuite.DELIMITER): String = {
-    val tokens = key.split(delimiter)
-    if (tokens.size != 2) {
-      throw new IllegalArgumentException(s"Provided input [$key] is not a valid feature key")
-    }
-    tokens(0)
+    require(delimiter.r.findAllIn(key).length == 1, s"Provided input [$key] is not a valid feature key")
```

Comment:
The original code had a bug: when the term is an empty string "", the size of the split array is 1, e.g.:

```scala
"name,".split(",")
// => res1: Array[String] = Array(name)
```

This causes the ```tokens.size != 2``` check to fail.